### DISCUSSION TITLE
Saving the Mesh after Parametrization

### DISCUSSION THREAD

**Discussion by AbhijithMoni** [mdolab/MACH-Aero#133] [Q&A] [answered]
It is a query regarding the Aerodynamic Optimization tutorial, specific to [Geometric Parametrization](https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/machAeroTutorials/opt_ffd.html). The tutorial briefly explains about how FFD box is created and visualizing the deformed FFD box. 

I would like to know if I can use IDWrap to save a parameterized grid file. Something like `wing_vol_parameterized.cgns`. It will be helpful to compare the original grid with the parameterized one.

I tried the tutorial available [here](https://mdolab-idwarp.readthedocs-hosted.com/en/latest/tutorial.html)  but was unsuccessful since it did not utilize the FFD box method.

Thanks.

**Comment by A-CGray** [comment]
Hi Abjith,

I'm not sure I understand your question, what do you mean by a "paramterized grid file"?

The idwarp tutorial you linked to doesn't assume you are using an FFD.

**Comment by AbhijithMoni** [comment]
Hi @A-CGray 

Thanks for your reply. I will rephrase my question with a visual example. Within the tools available in MDO lab packages, I am interested in understanding whether it is possible to directly parameterize a CGNS mesh using the FFD box, driven by design variable moments. And then save the parameterized mesh in CGNS format. 

**FFD box for parameterizing the mesh:**

![image](https://github.com/mdolab/MACH-Aero/assets/94369137/6fefb6d9-6fe5-41c8-947b-4a7b90633c85)

**Resulting mesh:**

![image](https://github.com/mdolab/MACH-Aero/assets/94369137/ea43c459-90f1-4b34-a2d3-b539a5af5a94)


Many thanks 

Best 
Abhijith

**Comment by A-CGray** [comment, marked as answer]
Yes this is possible. You'll have to combine the steps from the two tutorials you linked to.

Basically, follow the same steps as the idwarp tutorial you linked, but then instead of updating the surface coordinates using this code:

```python
# Modify the coordinates as required
newCoords = coords0.copy()
for i in range(len(coords0)):
    newCoords[i, :] *= 1.1
```

You will want to generate `newCoords` using the FFD following the steps in the MACH-Aero tutorial you linked to. The morphed surface coordinates you need to pass to idwarp are returned by the `DVGeo.update` function

**Reply by AbhijithMoni** [reply]
Thanks a lot for your input and time @A-CGray. Following the steps you provided, I was able to parameterise the CGNS grid.
